bundles/core/src/main/java/com/adobe/cq/commerce/core/search/internal/services/SearchResultsServiceImpl.java

```diff
@@ -181,9 +180,15 @@ public SearchResultsSet performSearch(
             request,
             resource);
 
-        final List<SearchAggregation> searchAggregations = extractSearchAggregationsFromResponse(products.getAggregations(),
+        List<SearchAggregation> searchAggregations = extractSearchAggregationsFromResponse(products.getAggregations(),
             mutableSearchOptions.getAllFilters(), availableFilters);
 
+        // Special treatment for category_id filter as this is always present and collides with category_uid filter (CIF-2206)
+        if (mutableSearchOptions.getCategoryUid().isPresent()) {
+            searchAggregations = searchAggregations.stream().filter(s -> !"category_id".equals(s.getIdentifier())).collect(Collectors
```

Comment:
can you use `Collection#removeIf(Predicate)` instead?

Reply:
+1 done